### PR TITLE
Replace tempdir with tempfile

### PR DESCRIPTION
**Pull request by lann**
`tempdir` is abandoned

### CODE REVIEW

File: crates/trigger/src/cli.rs [hunk 1]
```diff
@@ -312,7 +312,7 @@ impl<T: Trigger> TriggerAppBuilder<T> {
         let use_gpu = true;
         let state_dir = match options.state_dir {
             // Make sure `--state-dir=""` unsets the state dir
-            Some(s) if s.trim().is_empty() => None,
+            Some("") => None,
```
Comment (lann) [inline crates/trigger/src/cli.rs:315]: Sneaky lint fix...

**Review by kate-goldenring — APPROVED** [review]
LGTM! Thanks!

File: crates/trigger/src/cli.rs (hunk 1 above)
Comment (rylev) [reply crates/trigger/src/cli.rs:315]: This isn't equivalent, no? `Some("  ")` would match the first but not the second.

Comment (lann) [reply crates/trigger/src/cli.rs:315]: Ah sorry I blindly obeyed clippy. That said, `--state-dir " "` seems more likely to be an accident...

Comment (lann) [reply crates/trigger/src/cli.rs:315]: Though it now occurs to me that its perfectly valid to create a directory named `" "`...

**Pull request merged**